src/Language/PureScript/Bundle.hs

```diff
@@ -612,26 +672,39 @@ codeGen optionsMainModule optionsNamespace ms =  renderToString (JSAstProgram (p
-bundle :: (MonadError ErrorMessage m)
-       => [(ModuleIdentifier, String)] -- ^ The input modules.  Each module should be javascript rendered from 'Language.PureScript.Make' or @psc@.
+bundleSM :: (MonadError ErrorMessage m)
+       => [(ModuleIdentifier, String, String)] -- ^ The input modules.  Each module should be javascript rendered from 'Language.PureScript.Make' or @psc@.
        -> [ModuleIdentifier] -- ^ Entry points.  These module identifiers are used as the roots for dead-code elimination
        -> Maybe String -- ^ An optional main module.
        -> String -- ^ The namespace (e.g. PS).
-       -> m String
-bundle inputStrs entryPoints mainModule namespace = do
+       -> Maybe String -- ^ The output file name (if there is one - in which case generate source map)
+       -> m (Maybe SourceMapping, String)
+bundleSM inputStrs entryPoints mainModule namespace outFilename = do
+  let mid (a,_,_) = a
   forM_ mainModule $ \mname ->
-    when (mname `notElem` map (moduleName . fst) inputStrs) (throwError (MissingMainModule mname))
+    when (mname `notElem` map (moduleName . mid) inputStrs) (throwError (MissingMainModule mname))
   forM_ entryPoints $ \mIdent ->
-    when (mIdent `notElem` map fst inputStrs) (throwError (MissingEntryPoint (moduleName mIdent)))
-  input <- forM inputStrs $ \(ident, js) -> do
+    when (mIdent `notElem` map mid inputStrs) (throwError (MissingEntryPoint (moduleName mIdent)))
+  input <- forM inputStrs $ \(ident, filename, js) -> do
                 ast <- either (throwError . ErrorInModule ident . UnableToParseModule) pure $ parse js (moduleName ident)
-                return (ident, ast)
+                return (ident, filename, ast)
 
-  let mids = S.fromList (map (moduleName . fst) input)
+  let mids = S.fromList (map (moduleName . mid) input)
 
-  modules <- traverse (fmap withDeps . uncurry (toModule mids)) input
+  modules <- traverse (fmap withDeps . (\(a,fn,c) -> toModule mids a fn c)) input
 
   let compiled = compile modules entryPoints
       sorted   = sortModules (filter (not . isModuleEmpty) compiled)
 
-  return (codeGen mainModule namespace sorted)
+  return (codeGen mainModule namespace sorted outFilename)
+
+-- | The bundling function.
+-- This function performs dead code elimination, filters empty modules
+-- and generates and prints the final Javascript bundle.
+bundle :: (MonadError ErrorMessage m)
+       => [(ModuleIdentifier, String)] -- ^ The input modules.  Each module should be javascript rendered from 'Language.PureScript.Make' or @psc@.
+       -> [ModuleIdentifier] -- ^ Entry points.  These module identifiers are used as the roots for dead-code elimination
+       -> Maybe String -- ^ An optional main module.
+       -> String -- ^ The namespace (e.g. PS).
+       -> m String
+bundle inputStrs entryPoints mainModule namespace = snd <$> bundleSM (map (\(a,b) -> (a,"",b)) inputStrs) entryPoints mainModule namespace Nothing
```

Comment:
Not used in the no-source-maps code path, I guess this should be `Maybe`, or some better structuring generally